### ISSUE TITLE
Bugs about automatic regridding for cube-sphere when plotting

### ISSUE THREAD

**Issue by yuanjianz** [geoschem/gcpy#310]
### Name and Institution (Required)

Name: Yuanjian Zhang
Institution: WashU

### Confirm you have reviewed the following documentation

- [x] [Support guidelines](https://gcpy.readthedocs.io/en/stable/reference/SUPPORT.html)
- [x] [User manual](https://gcpy.readthedocs.io/en/stable)
- [x] [Current and past Github issues](https://github.com/geoschem/gcpy/issues)

### Description of your issue or question

Current main branch of GcPy seems to have trouble reformatting during the process of cube-sphere regridding.

1) Cube-sphere emissions and column AOD will trigger an error in the auto reformatting in `compare_single_level` functions due to lack of the dimension `lev`.
2) Auto regridding in `compare_single_level` for standard cube-sphere grids will only pick `ref` instead of finer resolution as documented.
3) Current regridding methods for cube-sphere grids will flipped `Xdim`, `Ydim` and `lon`, `lat`, which is causing horizontal flip when plotting single level in `single_panel` and `absdiff`, `fracdiff` in `compare_single_level`.

## Issue 1

When doing `benchmark_emis_plot` and `benchmark_aod_plot` between two cube-sphere files with different resolution, the program will trigger an error that says "cannot find `lev`" in `xarray.rename`.

## Issue 2 and 3

```python
compare_single_level(refds,'ref',devds,'dev',['SpeciesConcVV_DST1'])
```
![image](https://github.com/geoschem/gcpy/assets/84134417/5ebf0903-20d9-4c0c-b71f-32aae40e1024)
The function automatically chooses `c24` as comparing resolutiuon(issue 2), and we can also find the weird strips and incorrect negative bias in the figure(`dev`'s values should be larger than `ref`). The latter issue is caused by horizontal flip after regridding(issue 3), which can be confirmed by the figure following:

```python
single_panel(devds_regridded_from_c30_to_c24['SpeciesConcVV_DST1'].isel(lev=0,time=0))
```
* devds_regridded_from_c30_to_c24 is regridded from devds using file_regrid function following documentation
![image](https://github.com/geoschem/gcpy/assets/84134417/7018b9a1-be5b-4dd4-8f4a-26ded7434469)

**Comment by lizziel** [comment]
Thank you @yuanjianz! I think this addresses an old issue that was closed due to becoming stale: https://github.com/geoschem/gcpy/issues/167

**Comment by lizziel** [comment]
And this issue too: https://github.com/geoschem/gcpy/issues/147

**Comment by yuanjianz** [comment]
Hi @lizziel, I am convinced that these strips are caused by horizontal flip when plotting the regridded data. The reformatting process in regridding functions does not correctly reshape regridded files follwing the sequence of either `lat`,`lon` or `Ydim`, `Xdim`. This could be handled by `reshape_MAPL_cs` but I would rather the output from regridding process to be already in good shape. Please see the pull request #311.

**Comment by yuanjianz** [comment]
Also note that each cube-sphere face is flipped separately, thus typical horizontal flip features cannot be told from a global plot.

**Comment by yantosca** [comment]
We can close this issue as PR #311 has been merged into GCPy 1.5.0.
